base/linalg/symmetric.jl

```diff
@@ -26,7 +32,7 @@ convert{T,S<:AbstractMatrix}(::Type{Hermitian{T,S}},A::Hermitian) = Hermitian{T,
 convert{T}(::Type{AbstractMatrix{T}}, A::Hermitian) = Hermitian(convert(AbstractMatrix{T}, A.data), symbol(A.uplo))
 copy{T,S}(A::Symmetric{T,S}) = Symmetric{T,S}(copy(A.data),A.uplo)
 copy{T,S}(A::Hermitian{T,S}) = Hermitian{T,S}(copy(A.data),A.uplo)
-ishermitian(A::Hermitian) = true
+ishermitian(A::Hermitian) = all(isreal(diag(A.data)))
```

Comment:
but if a real diagonal is enforced during construction it is okay to assume that it is real at this point.